src/components/common/Map.js

```diff
@@ -1,26 +1,57 @@
-import React, {PropTypes} from 'react';
-import {GoogleMapLoader, GoogleMap, Marker} from 'react-google-maps';
+import React, { PropTypes } from 'react';
+import { GoogleMapLoader, GoogleMap, Marker } from 'react-google-maps';
 
 class Map extends React.Component {
-  constructor(props, context){
+  constructor(props, context) {
     super(props, context);
+
+    this.onMapClick = this.onMapClick.bind(this);
   }
 
-  render() {
-    const mapContainer = <div style={{position: 'absolute', height:'100%', width:'100%'}}></div>;
+  onMapClick(e) {
+    let lat = parseFloat(e.latLng.lat().toFixed(6));
+    let lng = parseFloat(e.latLng.lng().toFixed(6));
+    let marker = {
+      lat: lat,
+      lng: lng,
+      defaultAnimation: 2,
+    };
+    this.props.addMarker(marker);
+  }
 
-    return(
-      <GoogleMapLoader
-      containerElement = {mapContainer}
-      googleMapElement = {
-        <GoogleMap
-          defaultZoom = {15}
-          defaultCenter={{ lat: 46.2017559, lng: 6.1466014 }}
-          options={{streetViewControl: false, mapTypeControl: false}}>
-        </GoogleMap>
-      } />
+  render() {
+    const mapContainer = <div style={{ position: 'absolute', height: '100%', width: '100%' }} />;
+    const markers = this.props.markers;
```

Comment:
It's good practice to deconstruct the props objects when accessing its contents. 

`const { markers } = this.props;`

It's acceptable to use `this.props.markers` here because it's the only prop you use in this method, however, I think you should change it because in cases where you use more than one prop, it's better to have an statement like this one: `const { prop1, prop2, prop3 } = this.props` than writing `this.props.prop1` and thus every time.

Besides, for consistency, if you'll use that syntax when you have more than one prop, I'd recommend you use it everywhere for consistency.